lib/pbench/agent/redis.py

```diff
@@ -222,39 +222,76 @@ def emit(self, record):
             self.counter += 1
 
 
-def wait_for_conn_and_key(redis_server, key, prog, redis_host, redis_port):
+def wait_for_conn_and_key(redis_server: redis.Redis, key: str, prog: str) -> str:
     """wait_for_conn_and_key - convenience method of both the Tool Meister and
     the Tool Data Sink to startup and wait for an initial connection to the
     Redis server, and for the expected key to show up.
+
+    Arguments:
+
+        redis_server: a Redis client object
+        key: the key name as a string
+        prog: the program name of our caller
+
+    Returns the payload value of the key as a string.
+
+    If successful on the first connection attempt, and the value of the key is
+    retrieved, no messages are issued on stdout.  If the key does not exist yet,
+    we'll report the missing key every 10 seconds from the last successful
```

Comment:
This line is the answer to a question you asked elsewhere: "we'll report the missing key every 10 seconds". So, OK, we're each half right; the "code" does make assumptions about the time-span, but it's not _executable_ code nor visible to the end user. Then again, an outdated comment is less likely to trigger a unit test failure and get noticed when something changes, but can still cause confusion to future readers. ;-)